### PR TITLE
fix kubectl resources with all-namespace output format error

### LINKED ISSUE #10818: kubectl get rc output format error
The number of replicas are not properly lined up.
```
$ kubectl get rc --all-namespaces
NAMESPACE     CONTROLLER                     CONTAINER(S)                                     IMAGE(S)                                          SELECTOR                                   REPLICAS
kube-system   elasticsearch-logging-v1       elasticsearch-logging                            satnam6502/elasticsearch:1.7                      k8s-app=elasticsearch-logging,version=v1   2
kube-system   kibana-logging-v1              kibana-logging                                   gcr.io/google_containers/kibana:1.3               k8s-app=kibana-logging,version=v1          1
kube-system   kube-dns-v5                    etcd                                             gcr.io/google_containers/etcd:2.0.9               k8s-app=kube-dns,version=v5                1
              kube2sky                       gcr.io/google_containers/kube2sky:1.11                                                             
              skydns                         gcr.io/google_containers/skydns:2015-03-11-001                                                     
kube-system   kube-ui-v1                     kube-ui                                          gcr.io/google_containers/kube-ui:v1               k8s-app=kube-ui,version=v1         1
kube-system   monitoring-heapster-v5         heapster                                         gcr.io/google_containers/heapster:v0.15.0         k8s-app=heapster,version=v5        1
kube-system   monitoring-influx-grafana-v1   influxdb                                         gcr.io/google_containers/heapster_influxdb:v0.3   k8s-app=influxGrafana,version=v1   1
              grafana                        gcr.io/google_containers/heapster_grafana:v0.7                                                     
satnam@satnam-linux:~/gocode/src/github.com/GoogleCloudPlatform/kubernetes/contrib/logging/fluentd-sidecar-es$ kubectl version
Client Version: version.Info{Major:"0", Minor:"20+", GitVersion:"v0.20.2-421-g316fe730ff94b4-dirty", GitCommit:"316fe730ff94b4ac7d37b7b7f7d0fb00ca05d437", GitTreeState:"dirty"}
Server Version: version.Info{Major:"0", Minor:"20+", GitVersion:"v0.20.2-424-gd78358e229b04a-dirty", GitCommit:"d78358e229b04affef11c9fd95cd6feea9696497", GitTreeState:"dirty"}
```
@davidopp

### PR DESCRIPTION
**Pull request by jiangyaoguo**
fix #10818

### CODE REVIEW

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -476,10 +476,18 @@ func printPodTemplate(pod *api.PodTemplate, w io.Writer, withNamespace bool, wid
 
 	// Lay out all the other containers on separate lines.
 	for _, container := range containers {
-		_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", "", container.Name, container.Image, "")
-		if err != nil {
-			return err
+		if withNamespace {
+			_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\n", "", "", container.Name, container.Image, "")
```
Comment (satnam6502) [inline pkg/kubectl/resource_printer.go]: Minor, but you could put this in the if condition to make `err` local to the if statement.

**Comment by erictune** [comment]
thanks for fixing this.

**Comment by jiangyaoguo** [comment]
@satnam6502

**Comment by satnam6502** [comment]
Still LGTM from me. Any other action required? I think we need two TLs? @thockin @bgrant0607 perhaps?

**Comment by thockin** [comment]
LGTM, but not p0 so it has to wait.

**Comment by bgrant0607** [comment]
Reconcile with #10851. 

I have 0 time to look at this, but I'm fine with fixing the output if @jlowdermilk signs off on it.

**Comment by jbeda** [comment]
This looks great but I think my PR (#10851) is more comprehensive.  It has tests and also fixes alignment issues present when label columns are used.

**Comment by jlowdermilk** [comment]
@bgrant0607 if we are going to fix this output, #10851 is indeed more thorough (and has a test to boot).

**Comment by bgrant0607** [comment]
I delegate the decision to @jlowdermilk.

**Comment by jlowdermilk** [comment]
Lets merge #10851. Misaligned output is potentially very confusing.

@jiangyaoguo, thanks for your PR. Looks like @jbeda beat you to it with a fix/test in #10851, so we'll let that one supersede.

**Pull request closed without merging**